openedx/core/djangoapps/programs/utils.py

```diff
@@ -142,6 +170,11 @@ def progress(self, programs=None, count_only=True):
             list of dict, each containing information about a user's progress
                 towards completing a program.
         """
+        now = datetime.datetime.now(utc)
+        enrolled_run_modes = {}
+        for enrollment in self.enrollments:
+            enrolled_run_modes[unicode(enrollment.course_id)] = enrollment.mode
```

Comment:
Looking more closely, you should just replace `self.course_run_ids` with this dict. The elements in that list are the same as the keys in this dict. It's wasteful to loop over the enrollments twice to create a list and a dict with the same data.